src/h5web/visualizations/containers/NxImageContainer.tsx

```diff
@@ -40,34 +43,44 @@ function NxImageContainer(props: VisContainerProps): ReactElement {
     throw new Error('Expected signal dataset with at least two dimensions');
   }
 
-  const [mapperState, setMapperState] = useState<DimensionMapping>([
+  const [dimensionMapping, setDimensionMapping] = useState<DimensionMapping>([
     ...range(dims.length - 2).fill(0),
     'y',
     'x',
   ]);
 
-  const axes = getNxAxes(nxDataGroup, metadata);
-
-  const values = useDatasetValues({ signal: signalDataset.id, ...axes.ids });
+  const nxAxisMapping = getNxAxisMapping(nxDataGroup);
+  const axisDatasets = getDatasetsFromMapping(
+    nxDataGroup,
+    metadata,
+    nxAxisMapping
+  );
+  const datasetValues = useDatasetValues({
+    [signalName]: signalDataset.id,
+    ...Object.fromEntries(
+      Object.entries(axisDatasets).map(([axis, dataset]) => [axis, dataset.id])
+    ),
+  });
+  const nxAxisParams = getNxAxisParams(axisDatasets, datasetValues);
```

Comment:
I would just rename this `nxAxesParams` and `getNxAxesParams` for consistency with `axesParams`.

My reasoning is that `axisParams`, although technically correct, could be misinterpreted. Note that I'm happy with `axisMapping` and not `axesMapping`, though, because it's consistent with `dimensionMapping`.